File: leetcode/AIcoding/3.DeepSeek-v3/javascript_solutions/3158-find-the-xor-of-numbers-which-appear-twice.js
var duplicateNumbersXOR = function(nums) {
    const seen = new Set();
    let xor = 0;
    for (const num of nums) {
        if (seen.has(num)) {
            xor ^= num;
        } else {
            seen.add(num);
        }
    }
    return xor;
};

File: leetcode/AIcoding/1.Gemini-2.5/go_solutions/3158-find-the-xor-of-numbers-which-appear-twice.go
func duplicateNumbersXOR(nums []int) int {
	counts := make(map[int]int)
	for _, num := range nums {
		counts[num]++
	}

	xorSum := 0
	for num, count := range counts {
		if count == 2 {
			xorSum ^= num
		}
	}

	return xorSum
}